react/game/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,112 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.scss';
 import Board from './components/Board';
 
-function Game() {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [
+      {
+        squares: Array(9).fill(null)
+      }
+    ],
+    xIsNext: true,
+    stepNumber: 0,
+    winner: null
+  };
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (this.state.winner || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext,
+      winner: this.calculateWinner(squares)
+    });
+  };
+
+  jumpTo = step => {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0,
+      winner: step === this.state.history.length - 1 ? this.state.winner : null
+    });
+  };
+
+  checkStatus = () => {
+    if (this.state.winner) {
+      if (this.state.winner === 'tie') {
+        return `We have a TIE`;
+      }
+      return `Winner: ${this.state.winner}`;
+    }
+    return `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
+  };
+
+  calculateWinner = squares => {
+    const lines = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
```

Comment:
U don't need to declare this const in each function call. Extract this const in another file, in `src/constants/game.js`